libminifi/include/FlowController.h

```diff
@@ -198,125 +197,125 @@ class FlowController : public core::controller::ControllerServiceProvider, publi
-  virtual void clearControllerServices();
+  void clearControllerServices() override;
 
   /**
    * Gets all controller services.
    */
-  virtual std::vector<std::shared_ptr<core::controller::ControllerServiceNode>> getAllControllerServices();
+  std::vector<std::shared_ptr<core::controller::ControllerServiceNode>> getAllControllerServices() override;
 
-  virtual std::shared_ptr<core::controller::ControllerService> getControllerService(const std::string &identifier);
+  std::shared_ptr<core::controller::ControllerService> getControllerService(const std::string &identifier) override;
 
   /**
    * Gets controller service node specified by <code>id</code>
    * @param id service identifier
    * @return shared pointer to the controller service node or nullptr if it does not exist.
    */
-  virtual std::shared_ptr<core::controller::ControllerServiceNode> getControllerServiceNode(const std::string &id);
+  std::shared_ptr<core::controller::ControllerServiceNode> getControllerServiceNode(const std::string &id) override;
 
-  virtual void verifyCanStopReferencingComponents(std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode);
+  void verifyCanStopReferencingComponents(std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode) override;
 
   /**
    * Unschedules referencing components.
    */
-  virtual std::vector<std::shared_ptr<core::controller::ControllerServiceNode>> unscheduleReferencingComponents(std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode);
+  std::vector<std::shared_ptr<core::controller::ControllerServiceNode>> unscheduleReferencingComponents(std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode) override;
 
   /**
    * Verify can disable referencing components
    * @param serviceNode service node whose referenced components will be scheduled.
    */
-  virtual void verifyCanDisableReferencingServices(std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode);
+  void verifyCanDisableReferencingServices(std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode) override;
 
   /**
    * Disables referencing components
    * @param serviceNode service node whose referenced components will be scheduled.
    */
-  virtual std::vector<std::shared_ptr<core::controller::ControllerServiceNode>> disableReferencingServices(std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode);
+  std::vector<std::shared_ptr<core::controller::ControllerServiceNode>> disableReferencingServices(std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode) override;
 
   /**
    * Verify can enable referencing components
    * @param serviceNode service node whose referenced components will be scheduled.
    */
-  virtual void verifyCanEnableReferencingServices(std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode);
+  void verifyCanEnableReferencingServices(std::shared_ptr<core::controller::ControllerServiceNode>&) override;
```

Comment:
As far as I understand, this function doesn't do anything.

Reply:
I wanted to limit the scope of my extra changes to the originally affected file. Since this is an overridden virtual function, I didn't reason about possible changes here.

From what I can see, the class has a `ControllerServiceProvider` base and a member, and the overridden function delegates to the same function of the member instead of the base.